### PR TITLE
Make it possible to recover from safemode without restarting

### PR DESCRIPTION
**Pull request by Warriorrrr**
#### Description: 

- Important methods that should always be ran and which don't throw TownyInitExceptions themselves have been moved out of the try block.
- Most importantly, commands are now always registered which makes it possible to use /ta reload again.
- The list of towny errors has been changed to a set to ensure that an error can only be added once, and that using removeError will always make a subsequent isError for the same error will always be false.
- Checking for third party plugins now always happens because the permission source would otherwise be null. If the main config does not load correctly economy will be enabled as soon as the config successfully loads.
- Improved the error logging when something does not /ta reload correctly, if something isn't reloading correctly it'll now enable safemode.
- Always register all listeners regardless of whether we're in safemode or not, many of the listeners already had safemode checks.
____
#### New Nodes/Commands/ConfigOptions: 

____
#### Relevant Towny Issue ticket:

____
- [ ] I have tested this pull request for defects on a server. 

By making this pull request, I represent that I have the right to waive copyright and related rights to my contribution, and agree that all copyright and related rights in my contributions are waived, and I acknowledge that the TownyAdvanced organization has the copyright to use and modify my contribution under the Towny [License](https://github.com/LlmDl/Towny/blob/master/LICENSE.md) for perpetuity.

### CODE REVIEW

**Comment by Warriorrrr** [comment]
All of the commands (other than townyadmin) all check for whether safemode is enabled to prevent the things described above, the only danger might be with addon plugins accidentally triggering stuff. The internal annotation could be removed from the various isError methods so that addon plugins can check whether various areas loaded correctly and account for that.

**Comment by LlmDl** [comment]
Ah you are correct, I dont think any addon-command-api users should be getting their commands sent over though.

**Comment by Warriorrrr** [comment]
I was thinking more about plugins that integrate with Towny and add their own commands, those could still be doing lookups or saving.

File: Towny/src/main/java/com/palmergames/bukkit/towny/command/TownyAdminCommand.java [hunk 1]
```diff
@@ -2088,31 +2097,35 @@ public void reloadConfig(CommandSender sender, boolean reset) {
...
 			TownySettings.loadTownLevelConfig();   // TownLevel and NationLevels are not loaded in the config,
 			TownySettings.loadNationLevelConfig(); // but later so the config-migrator can do it's work on them if needed.
-			Translation.loadTranslationRegistry();
+			plugin.loadLocalization(true);
 			TownBlockTypeHandler.initialize();
-		} catch (TownyException e) {
+
+			TownyMessaging.sendMsg(sender, Translatable.of("msg_reloaded_config"));
+			plugin.removeError(TownyInitException.TownyError.MAIN_CONFIG);
+			plugin.removeError(TownyInitException.TownyError.LOCALIZATION);
```
Comment (LlmDl) [inline Towny/src/main/java/com/palmergames/bukkit/towny/command/TownyAdminCommand.java:2108]: These two are already being called when we're in plugin#loadConfig and plugin#loadLocalization, do we need to call them again?

Comment (Warriorrrr) [reply Towny/src/main/java/com/palmergames/bukkit/towny/command/TownyAdminCommand.java]: My sources say no

**Pull request merged**